fairscale/nn/moe/top2gate.py

```diff
@@ -71,8 +64,8 @@ def top2gating(logits: torch.Tensor) -> Tuple[Tensor, Tensor, Tensor]:
     # Normalize gate probabilities
     mask1_float = mask1.float()
     mask2_float = mask2.float()
-    gates1_gs = torch.einsum("gse,gse->gs", gates, mask1_float)
-    gates2_gs = torch.einsum("gse,gse->gs", gates, mask2_float)
+    gates1_gs = torch.einsum("gse,gse->gs", torch.exp(logits), mask1_float)
+    gates2_gs = torch.einsum("gse,gse->gs", torch.exp(logits), mask2_float)
```

Comment:
will torch be smart enough and perform a single torch.exp on logits here or it will be done twice on the same data?